pkgs/development/tools/misc/nimlsp/default.nix

```diff
@@ -0,0 +1,58 @@
+{ stdenv, fetchFromGitHub, nim, termbox, pcre }:
+
+let
+  astpatternmatching = fetchFromGitHub {
+    owner = "krux02";
+    repo = "ast-pattern-matching";
+    rev = "87f7d163421af5a4f5e5cb6da7b93278e6897e96";
+    sha256 = "19mb5bb6riia8380p5dpc3q0vwgrj958dd6p7vw8vkvwiqrzg6zq";
+  };
+
+  jsonschema = fetchFromGitHub {
+    owner = "PMunch";
+    repo = "jsonschema";
+    rev = "7b41c03e3e1a487d5a8f6b940ca8e764dc2cbabf";
+    sha256 = "1js64jqd854yjladxvnylij4rsz7212k31ks541pqrdzm6hpblbz";
+  };
+
+  nim-src = fetchFromGitHub {
```

Comment:
Is there a way you can reuse https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/nim/default.nix ?

Reply:
Yea, this could be expressed as `nim-src = nim.src;`.